src/azure-cli-core/azure/cli/core/auth/adal_authentication.py

```diff
@@ -18,9 +19,25 @@ class MSIAuthenticationWrapper(MSIAuthentication):
     def get_token(self, *scopes, **kwargs):  # pylint:disable=unused-argument
         logger.debug("MSIAuthenticationWrapper.get_token invoked by Track 2 SDK with scopes=%s", scopes)
 
+        # In Cloud Shell, we use
+        # - msrestazure to get access token
+        # - MSAL to get VM SSH certificate
         if 'data' in kwargs:
-            from azure.cli.core.azclierror import AuthenticationError
-            raise AuthenticationError("VM SSH currently doesn't support managed identity or Cloud Shell.")
+            if in_cloud_console():
```

Comment:
@jiasli, do we need more sophisticated logic here? If the user runs an explicit `az login` and login successfully, this `in_cloud_console()` would still return `True` and MSAL Python would still get an ssh cert for "current user", but perhaps we would want to run the normal ssh cert code path in order to get a ssh cert for the explicitly-logged-in user?

Reply:
No worry. `MSIAuthenticationWrapper` will not even be created for explicitly-logged-in user. It is guarded by `get_login_credentials`:

https://github.com/Azure/azure-cli/blob/d331edf76e5866cda1e6e9641d935256fc4f03ff/src/azure-cli-core/azure/cli/core/_profile.py#L319-L321